### DISCUSSION TITLE
Reintroduce actors

### DISCUSSION THREAD

**Discussion by 0bon** [scala/scala3#18666] [Feature Requests]
In regards to the licensing fiasco with Akka and lightbend, it would be great to bring scala actors back from the dead as an alternative to the akka eco-system. Please discuss.

**Comment by som-snytt** [comment]
Merits a discussion! https://github.com/lampepfl/dotty/discussions

If they are provided by pekko, are actors called pectors?

**Comment by jducoeur** [comment]
Scala actors were abandoned in no small part because they were already awfully primitive.  (I built a system on top of them, but it was very limited.)  At this point, they're hopelessly behind the times.

The right alternative to Akka, for those who are looking for one, is probably Pekko, which starts out from a mature and reasonably state of the art basis.

**Comment by SethTisue** [comment]
If interested in something lighter weight than Akka/Pekko, there's also https://github.com/com-lihaoyi/castor

in Typelevel ecosystem, see discussion at https://github.com/typelevel/cats-effect/discussions/2627

**Comment by jducoeur** [comment]
Speaking of Typelevel, if you're looking for event-sourcing (which as far as I'm concerned was always the killer app of Akka), there's also Edomata: https://github.com/hnaderi/edomata

**Comment by plokhotnyuk** [comment]
The smallest and the most efficient actor is [here](https://github.com/plokhotnyuk/actors/blob/master/src/test/scala/com/github/gist/viktorklang/Actor.scala)

**Comment by som-snytt** [comment]
Our smallest and most efficient actor is [here](https://en.wikipedia.org/wiki/Peter_Dinklage)
